akka-remote/src/test/scala/akka/remote/artery/LargeMessagesChannelSpec.scala

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import akka.actor.{ Actor, ActorRef, ActorSelection, ActorSystem, ExtendedActorSystem, Props, RootActorPath }
+import akka.remote.{ LargeDestination, RegularDestination, RemoteActorRef }
+import akka.testkit.{ SocketUtil, TestProbe }
+import akka.util.ByteString
+import com.typesafe.config.ConfigFactory
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ ShouldMatchers, WordSpec }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+object LargeMessagesChannelSpec {
+  case class Ping(payload: ByteString = ByteString.empty)
+  case class Pong(bytesReceived: Long)
+  class EchoSize extends Actor {
+    def receive = {
+      case Ping(bytes) ⇒ sender() ! Pong(bytes.size)
+    }
+  }
+}
+
+class LargeMessagesChannelSpec extends WordSpec with ShouldMatchers with ScalaFutures {
+  import LargeMessagesChannelSpec._
+
+  def config(port: Int) = ConfigFactory.parseString(
+    s"""
+      akka {
+        loglevel = ERROR
+        actor {
+          provider = "akka.remote.RemoteActorRefProvider"
+        }
+        remote.artery {
+          enabled = on
+          hostname = localhost
+          port = $port
+          large-message-destinations = [
+            "/large"
+          ]
+        }
+      }
+
+    """)
+
+  "The large message support" should {
+
+    "not affect regular communication" in {
+      val Seq(portA, portB) = SocketUtil.temporaryServerAddresses(2, "localhost", udp = true).map(_.getPort)
+      val systemA = ActorSystem("systemA", config(portA))
+      val systemB = ActorSystem("systemB", config(portB))
+
+      try {
+        val senderProbeA = TestProbe()(systemA)
+        val senderProbeB = TestProbe()(systemB)
+
+        // start actor and make sure it is up and running
+        val large = systemB.actorOf(Props(new EchoSize), "regular")
+        large.tell(Ping(), senderProbeB.ref)
+        senderProbeB.expectMsg(Pong(0))
+
+        // communicate with it from the other system
+        val addressB = systemB.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+        val rootB = RootActorPath(addressB)
+        val largeRemote = awaitResolve(systemA.actorSelection(rootB / "user" / "regular"))
+        largeRemote.tell(Ping(), senderProbeA.ref)
+        senderProbeA.expectMsg(Pong(0))
+
+        // flag should be cached now
+        largeRemote.asInstanceOf[RemoteActorRef].cachedLargeMessageDestinationFlag should ===(RegularDestination)
+
+      } finally {
+        systemA.terminate()
+        systemB.terminate()
+      }
+    }
+
+    "pass small regular messages over the large-message channel" in {
+      val Seq(portA, portB) = SocketUtil.temporaryServerAddresses(2, "localhost", udp = true).map(_.getPort)
```

Comment:
Ah, yes, I'm anyways extracting the address from the `ExtendedActorSystem`, started out like this because I didn't think of doing that and thought I needed to know the port number to create the address.